src/ompl/geometric/planners/experience/LightningRetrieveRepair.h

```diff
@@ -0,0 +1,202 @@
+        /// @endcond
+
+        /** \class ompl::base::LightningRetrieveRepairPtr
+            \brief A boost shared pointer wrapper for ompl::base::LightningRetrieveRepair */
+
+        /**
+           @anchor LightningRetrieveRepair - Lightning Retrieve Repair
+           @par Short description
+           LightningRetrieveRepair is a experienced-based motion planner that recalls from a database of
+           previous actions the most similar one to the current planning problem and attempts to repair it
+           @par External documentation
+           Berenson, Dmitry, Pieter Abbeel, and Ken Goldberg: A robot path planning framework that learns from experience, in <em>Robotics and Automation (ICRA), 2012 IEEE International Conference on. IEEE</em>, 2012.
+           <a href="http://users.wpi.edu/~dberenson/lightning.pdf">[PDF]</a>
+        */
+
+        /** \brief The Lightning Framework's Retrieve-Repair component */
+        class LightningRetrieveRepair : public base::Planner
+        {
+        public:
+
+            /** \brief Constructor */
+            LightningRetrieveRepair(const base::SpaceInformationPtr &si, const ompl::tools::LightningDBPtr &experienceDB);
+
+            virtual ~LightningRetrieveRepair(void);
+
+            /** \brief Get information about the exploration data structure the planning from scratch motion planner used. */
+            virtual void getPlannerData(base::PlannerData &data) const;
+
+            /**
+             *  \brief Get debug information about the top recalled paths that were chosen for further filtering
+             *  \return data - vector of PlannerData objects that each hold a single path
+             */
+            const std::vector<base::PlannerDataPtr>& getLastRecalledNearestPaths() const;
+
+            /**
+             *  \brief Get debug information about the top recalled paths that were chosen for further filtering
+             *  \return chosenID - the index of the PlannerData object that was chosen for repair
+             */
+            const std::size_t& getLastRecalledNearestPathChosen() const;
+
+            /**
+             * \brief Get the chosen path used from database for repair
+             * \return PlannerData of chosen path
+             */
+            ompl::base::PlannerDataPtr getChosenRecallPath() const;
+
+            /** \brief Get information about the exploration data structure the repair motion planner used each call. */
+            void getRepairPlannerDatas(std::vector<base::PlannerDataPtr> &data) const;
+
+            /** \brief See base class */
+            virtual base::PlannerStatus solve(const base::PlannerTerminationCondition &ptc);
+
+            /** \brief See base class */
+            virtual void clear(void);
+
+            /**
+             * \brief Pass a pointer of the database from the lightning framework
+             */
+            void setLightningDB(ompl::tools::LightningDBPtr &experienceDB);
```

Comment:
done. just to be sure, this means that the shared_ptr is const, not the value inside, right?